polar-core/src/validations.rs

```diff
@@ -218,16 +166,10 @@ impl ResourceBlocksMissingHasPermissionVisitor {
 
     fn warnings(&mut self) -> Option<Diagnostic> {
         if !self.calls_has_permission {
-            return Some(Diagnostic::Warning("Warning: your policy uses resource blocks but does not call the \
-has_permission rule. This means that permissions you define in a \
-resource block will not have any effect. Did you mean to include a \
-call to has_permission in a top-level allow rule?
-
-  allow(actor, action, resource) if
-      has_permission(actor, action, resource);
-
-For more information about resource blocks, see https://docs.osohq.com/any/reference/polar/polar-syntax.html#actor-and-resource-blocks".to_string()
-            ));
+            return Some(Diagnostic::Warning(PolarWarning {
+                kind: ValidationWarning::MissingHasPermissionRule,
+                context: None,
+            }));
```

Comment:
@dhatch this is the lone instance of manually constructing a `PolarWarning` that I mentioned on the call this morning. We know the `MissingHasPermissionRule` won't ever have source context and we don't have access to a KB in this context, so rather than pass a KB in I decided to manually construct the warning w/o context.

Reply:
Wouldn't the typical approach like with errors to be to return this as a ValidationWarning instead? And do the wrapping elsewhere. This is totally unimportant, just curious if there is a distinction between how we treat warnings

Reply:
derp; thanks @samscott89 

6e9dcea6